tool/src/org/antlr/v4/codegen/target/TypeScriptTarget.java

```diff
@@ -142,6 +144,10 @@ protected STGroup loadTemplates() {
 		STGroup result = new STGroupFile(CodeGenerator.TEMPLATE_ROOT+"/TypeScript/"+getLanguage()+STGroup.GROUP_FILE_EXTENSION);
 		result.registerRenderer(Integer.class, new NumberRenderer());
 		result.registerRenderer(String.class, new StringRenderer());
+		result.registerRenderer(
+			org.antlr.v4.codegen.model.Action.class, 
+			new TypeScriptActionRenderer()
+			);
```

Comment:
@sharwell this does not seem to be working as I expected.   The TypeScriptActionRenderer is not getting invoked.   Can you help me understand why?

Reply:
@sharwell, this PR may have been overcome by events, but it showed me I don't the Java code for multiple target support.   Can you have a look and tell me why TypeScriptActionRenderer doesn't get invoked on this branch?

Reply:
I'm actually not sure. I've never tried to add a renderer for that type.